python/dgl/distributed/optim/pytorch/sparse_optim.py

```diff
@@ -35,16 +45,168 @@ def __init__(self, params, lr):
             self._rank = 0
             self._world_size = 1
 
+    def local_state_dict(self):
+        """Return the state pertaining to current rank of the optimizer.
+
+        Returns
+        -------
+        dict
+            Local state dict
+
+        See Also
+        --------
+        load_local_state_dict
+        """
+        lcoal_state_dict = {}
+        lcoal_state_dict["emb_states"] = {}
+        num_clients = dgl.distributed.get_num_client()
```

Comment:
should use `role.get_num_trainers()`. `get_num_client()` contains trainers and sampler processes. Pls add test for it to make sure it can handle such scenario

Reply:
yes, and dgl distributed contains more and more complicated, to make it clear and more robust, use torch distributed instead.